jax/api.py

```diff
@@ -1743,6 +1744,16 @@ def _vjp_pullback_wrapper(cotangent_dtypes, io_tree, fun, py_args):
   return tree_unflatten(out_tree, ans)
 
 
+@overload
```

Comment:
I could add a TODO here about using PEP 586's `Literal` on `has_aux` to make this more ergonomic when JAX drops Ptyhon 3.7 one day.